### DISCUSSION TITLE
Engenharia de Software no desenvolvimento WEB

### DISCUSSION THREAD

**Discussion by Adejair** [frontendbr/forum#581] [Discussão]
Olá galera, tudo bem?

Eu sou um graduando do curso de Engenharia de Software e também sou desenvolvedor front/back de uma empresa. Estou com uma dúvida em relação a arquitetura que vocês usam para o desenvolvimento de um projeto. 

* Analise de Requisitos
* Proposta de Software
* Calendário de desenvolvimento

Se tivermos a oportunidade de discutir sobre isso, e tivermos exemplos práticos de como funciona, vai ser muito interessante, desde já, obrigado !

**Comment by felipefialho** [comment]
@mtmr0x e @Mendrone Sobre o que conversamos outro dia

**Comment by mtmr0x** [comment]
@Adejair Eu acho que tu foi um pouco generico ou eu entendi um pouco errado. rs

Tu quer falar sobre arquitetura de um software ou sobre as premissas (requisitos e proposta). E calendario de desenvolvimento ja é mais outro ponto, pois se encaixa em qual processo adotará para desenvolver a parada.

Consegue ser mais claro sobre a discussão pra nós? :smile:

**Comment by Adejair** [comment]
As premissas mesmo, gostaria de saber como vocês trabalham em relação a isso, se vocês tem exemplos ou algo do genero para poder apresentar, sou muito curioso nessa área.

**Comment by mtmr0x** [comment]
@Adejair olha, eu acho que a discussão seria mais simples do que imagina. Vou dar exemplo de um proxy de autorização que temos estudado refazer aqui.

Todo request seria batido nele primeiro para validar a sessão do usuário, logo a quantidade de requests nele seria bem alta, pois ele faria uma camada em cima de endpoints que necessitam de permissão.

- A tecnologia precisa ser de baixissima latencia;
- Linguagem multithreading e/ou multiprocessing
- Maturo gerenciamento multithreading e/ou multiprocessing em larga escala;
- Maturos logs para administrar qualquer anomalia ou tentativa de quebra de segurança;

E isso seria um draft pensando num cenário especifico, por isso disse que a discussão pode ser mais simples do que espera, é tudo muito relativo as particularidades do serviço que está criando.
